serverless/serverless
# #4750 Java invoke local support for handlers that implement RequestStreamHandler

## What did you implement:

Closes #4750 



## How did you implement it:
I performed some checks inside the invoke method to support `RequestStreamHandler` compatible handlers. The checks include testing whether or not the first parameter was an `InputStream` type, and also a similar check during the reflected method invocation.



## How can we verify it:
Unit tests are included that verify that `RequestStreamHandler` methods can receive input, and write to an `OutputStream`. You can verify it by running the tests from your IDE, or simply run `mvn clean test` in the root of the plugin's directory.

You can also run `mvn clean package` from the plugin's directory; it will run the tests, and output the plugin JAR to `target/invoke-bridge-1.0.jar`. You can take this JAR file and put it in your global `node_modules` directory that houses `serverless`. For instance on my machine that location is `/usr/local/lib/node_modules/serverless/lib/plugins/aws/invokeLocal/java/target`. Be sure to back up the original `serverless` JAR, just in case ;)

After replacing your JAR, you can run `sls invoke local` with your function (one that handles `RequestStreamHandler`), and provide a path to some target data file to verify functionality.



## Todos:


***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
